### ISSUE TITLE
Assert: "tree_index >= 0" after #1630 when model with $nocollide_this_only explodes

### ISSUE THREAD

**Issue by tomimaki** [scp-fs2open/fs2open.github.com#1708]
Release build gives me multiple times
![nocollide](https://user-images.githubusercontent.com/18072803/39009055-d45342a8-440a-11e8-8d5d-4960a28103c9.png)
and debug build: ASSERTION: "tree_index >= 0" at modelread.cpp:5232
[fs2_open.log](https://github.com/scp-fs2open/fs2open.github.com/files/1929204/fs2_open.log)

Example is Shivan Commnode from Mediavps 2014. Load mission Into Lion's Den (loop2-2.fs2), pack Trebuchet, destroy Crystal in Commnode and wait until it explode.

Another example is Vishnan_Protector from BP but can't find reliable mission.

**Comment by ngld** [comment]
The debug assertion is caused by the fact that [`submodel_get_two_random_points_better`](https://github.com/scp-fs2open/fs2open.github.com/blob/master/code/model/modelinterp.cpp#L1327) assumes that detail0 always has a collision tree. If the tree hasn't been loaded, the `tree_index` is `-1` which then causes that assertion in `model_get_bsp_collision_tree`.

A nice solution would be to make `submodel_get_two_random_points_better` use a collide subobject if one is available instead of detail0 but I don't know enough about the code to do this myself.

**Comment by Goober5000** [comment]
I don't know about collision subobjects myself, but it's a simple matter to fall back to the old method, so I made a new PR to do just that.

**Comment by tomimaki** [comment]
I can confirm that #1733 fixed issue.
